### ISSUE TITLE
Option to override source code location

### ISSUE THREAD

**Issue by Maxxan** [OpenCppCoverage/OpenCppCoverage#11]
We have a build server that compiles the application, meaning that progam debug database points to the location where the build server has the source code, like d:\tfs\myproject.

Now if I want to test the coverage, I set the `module path` to where the compiled project has been moved to, like \\networkdisk\myproject\relese, check out the source code to c:\my-tfs\myproject. and set `sources` to point to that location But since the path of the source code is different from the pdb files, OpenCppCoverage skips all source code since it doesn't match the pattern.

Is there any workaround to have the source code in a different location then when it was built?

**Comment by OpenCppCoverage** [comment]
Hello,

Unfortunately, I do not know a workaround. This feature is already in the [roadmap]( https://github.com/OpenCppCoverage/OpenCppCoverage/wiki/Roadmap#define-the-location-of-the-source-files) and should be available in the next release (Release date is not yet defined).

I hope it will help,

OpenCppCoverage

**Comment by Maxxan** [comment]
Great!
I hope the release isn't too far away in the future then ;)

**Comment by mladenmilev** [comment]
Hello, I am dealing with "similar situation", though I have some flexibility to move things around. Unfortunately I still could not make a single successful run of coverage. 

My Setup is the following:
VisualStudio C++ 11, about 10 dll. The project uses MFC. 
Unit tests with GTest framework. 
I recompiled those VS Solution in Debug - sources and tests.

```
Solution
   + Project A (sources)
        +--- Debug (holds A.pdb and A.dll)
   + Project B (sources)
        +--- Debug (holds B.pdb and B.dll)
   + Project C (sources)
        +--- Debug (holds C.pdb and C.dll)
   + UnitTests
        + ProjectA_UnitTests (sources UnitTests for A.dll)
        + ProjectB_UnitTests (sources UnitTests for B.dll)
        + ProjectC_UnitTests (sources UnitTests for C.dll)
        + Debug (All Project*_UnitTests.exe, A.dll, A.pdb, .... everything in same directory)
     
```
I can run with success in command line ALL tests executables Project*UnitTests.exe from C:\Solution\UnitTests\Debug

But I am failing to have coverage report. Ther HTML report is always empty.

Currently I run the following command:

`OpenCppCoverage.exe --modules=C:\Solution\UnitTests\Debug\A.dll --sources=C:\Solution\ProjectA\ -- C:\Solution\UnitTests\Debug\ProjectA_UnitTests.exe`

In the console I see it loads the files, runs the google tests. **Tests pass green and with success**,  then console waits for key stroke (do not know why? Could be the google test framewokr blocking it). Once I give a key stroke generates the html report but it is empty table html with the name of the ProjectA_UnitTests.exe.

Any idea what I am doing worng ? Seems it cannot "hook" to follow the execution
Do I need to have the ProjectA sources, A.dll, A.pdb, ProjectA_UnitTests.exe in same directory to make it work? 

Ideally I want ot have coverage for all Projects but I am trying to start with one. 

Any suggestions are welcomed. Hopefully the documentation how to run the tool can be further completed to describe those setups as I am struggling already few hours to make a coverage.

Thanks in advance.

**Comment by OpenCppCoverage** [comment]
Hello,

_¨Do I need to have the ProjectA sources, A.dll, A.pdb, ProjectA_UnitTests.exe in same directory to make it work?"_
_A.dll_ and _ProjectA_UnitTests.exe_ does not need to be in the same folder. _A.pdb_ should be in the same folder as A.dll if you move these 2 files after the compilation.

Can you try the following command line?
``
OpenCppCoverage.exe --sources=ProjectA -- C:\Solution\UnitTests\Debug\ProjectA_UnitTests.exe
``

If the report is still empty then you can try :

``
OpenCppCoverage.exe -- C:\Solution\UnitTests\Debug\ProjectA_UnitTests.exe
``
Note: The previous command may take some time to run but it is already fixed for the next release.

If the report is still empty could you provide the log file with --verbose flag?


I hope it will help,

OpenCppCoverage

**Comment by mladenmilev** [comment]
Ok, I got it working. First problem was related with the paths as I was calling on windows directly c:\Program Files\OpenCppCoverage.exe ... and seems did not worked well. After putting it in System path and reboot (I will later need to execute it as service with no logon) the coverage started working wirh same command.

Additional note: seems that if "modules" are defined then there is a need for module definition also for the path towards the UnitTests.exe. Else it is ignored! 

It was required to add the following line in my case:

`modules=workspace\UnitTests\Debug\`

So the config file looks like:

```
sources=workspace\ProjectA\
sources=workspace\ProjectB\
sources=workspace\ProjectC\
modules=workspace\ProjectA\Debug\
modules=workspace\ProjectB\Debug\
modules=workspace\ProjectC\Debug\
modules=workspace\UnitTests\Debug\
excluded_modules=workspace\dist\
excluded_modules=workspace\ExternalLibs\
export_type=html
```

Then I call:

`OpenCppCoverage.exe  --config_file=config.txt -- workspace\UnitTests\Debug\UnitTest.exe`

My next step is to figure a way to run multiple UnitTests.exe. 
One option is to run them one by one, produce binary output and join the results on last call. 

Any other options or suggestions?

**Comment by OpenCppCoverage** [comment]
Hello,

Your guess is correct. You can run one by one and then join the result. See [here](https://github.com/OpenCppCoverage/OpenCppCoverage/wiki#merging-coverage---input_coverage----export_typebinary) for more information.

I hope it will help,

OpenCppCoverage

**Comment by OpenCppCoverage** [comment]
Hello,

It should be fixed in [OpenCppCoverage 0.9.7.0](https://github.com/OpenCppCoverage/OpenCppCoverage/releases/tag/release-0.9.7.0).

OpenCppCoverage
